pkg/ccl/backupccl/generative_split_and_scatter_processor.go

```diff
@@ -260,8 +262,10 @@ func newGenerativeSplitAndScatterProcessor(
 
 	mkSplitAndScatterer := func() (splitAndScatterer, error) {
 		if spec.ValidateOnly {
-			nodeID, _ := flowCtx.NodeID.OptionalNodeID()
-			return noopSplitAndScatterer{nodeID}, nil
+			rng, _ := randutil.NewPseudoRand()
+			idx := rng.Intn(numNodes)
+			nodeID := spec.NodeIds[idx]
+			return noopSplitAndScatterer{roachpb.NodeID(nodeID)}, nil
```

Comment:
this still hard codes the node we'll use for all scatters. I think you need to pass `rng` to the `noopsplitscatterer` and call `rng.Intn()` every time it calls scatter.